### PR TITLE
Added RlistAppendString utility function to libpromises/rlist

### PR DESCRIPTION
**Pull request by oleorhagen**

### CODE REVIEW

**Comment by oleorhagen** [comment]
Ref: https://github.com/cfengine/core/pull/3870#discussion_r332402905

File: libpromises/rlist.c [hunk 1]
```diff
@@ -542,6 +542,15 @@ Rlist *RlistAppendScalar(Rlist **start, const char *scalar)
     return RlistAppendRval(start, RvalCopyScalar((Rval) { (char *)scalar, RVAL_TYPE_SCALAR }));
 }
 
+Rlist *RlistAppendString(Rlist **start, const char *string)
```
Comment (olehermanse) [inline libpromises/rlist.c]: [This does not represent a user facing change, and thus it should not have a changelog entry](https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#changelog). Please update commit message.

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2999/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2999/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2999/

**Pull request merged**